src/sinks/uds.rs

```diff
@@ -43,40 +45,63 @@ impl UdsMetricSink {
     /// # Example
     ///
     /// ```no_run
-    /// use std::os::unix::net::UnixStream;
+    /// use std::os::unix::net::UnixDatagram;
     /// use cadence::UdsMetricSink;
     ///
-    /// let socket = UnixStream::connect("/tmp/sock").unwrap();
-    /// let sink = UdsMetricSink::from(socket);
+    /// let socket = UnixDatagram::unbound().unwrap();
+    /// let sink = UdsMetricSink::from(socket, "/tmp/sock");
     /// ```
     ///
     /// To send metrics over a non-blocking socket, simply put the socket
     /// in non-blocking mode before creating the UDS metric sink.
     ///
     /// # Non-blocking Example
     ///
-    /// Note that putting the UDS socket into non-blocking mode is the
-    /// default when sink and socket are automatically created with the
-    /// `StatsdClient::from_uds_path` method.
-    ///
     /// ```no_run
-    /// use std::os::unix::net::UnixStream;
+    /// use std::os::unix::net::UnixDatagram;
     /// use cadence::UdsMetricSink;
     ///
-    /// let socket = UnixStream::connect("/tmp/sock").unwrap();
+    /// let socket = UnixDatagram::unbound().unwrap();
     /// socket.set_nonblocking(true).unwrap();
-    /// let sink = UdsMetricSink::from(socket);
+    /// let sink = UdsMetricSink::from(socket, "/tmp/sock");
     /// ```
-    pub fn from(socket: UnixStream) -> UdsMetricSink {
+    pub fn from<P: AsRef<Path>>(socket: UnixDatagram, path: P) -> UdsMetricSink {
         UdsMetricSink {
-            socket: Mutex::new(socket),
+            socket: socket,
+            path: path.as_ref().to_path_buf(),
         }
     }
 }
 
 impl MetricSink for UdsMetricSink {
     fn emit(&self, metric: &str) -> io::Result<usize> {
-        self.socket.lock().unwrap().write(metric.as_bytes())
+        self.socket.send_to(metric.as_bytes(), self.path.as_path())
+    }
+}
+
+/// Adapter for writing to a `UdsSocket` via the `Write` trait
```

Comment:
`UdsSocket` here should be `UnixDatagram`, right?

Reply:
Fixed